x-pack/plugins/maps/public/classes/layers/tiled_vector_layer/tiled_vector_layer.tsx

```diff
@@ -92,7 +93,34 @@ export class TiledVectorLayer extends VectorLayer {
 
     startLoading(SOURCE_DATA_REQUEST_ID, requestToken, searchFilters);
     try {
-      stopLoading(SOURCE_DATA_REQUEST_ID, requestToken, dataRequest, {});
+      let urlToken;
```

Comment:
Nit: How about simplifying lines 97 through 115 into the below statement to remove all the nested branches?

```
const prevMeta = prevDataRequest ? prevDataRequest.getMeta() : undefined;
const prevData = prevDataRequest ? prevDataRequest.getData() as MVTSingleLayerVectorSourceConfig : undefined;
const urlToken = !prevData || isRefreshOnlyQuery(prevMeta ? prevMeta.query : undefined, searchFilters.query)
  ? uuid()
  : prevData.urlToken;
```